clis/teliod/src/daemon.rs

```diff
@@ -0,0 +1,66 @@
+//! Code for general daemon handling utilities and abstracting away related dependencies.
+
+use std::path::PathBuf;
+
+use anyhow::Result;
+use sysinfo::{ProcessRefreshKind, RefreshKind, System};
+
+/// Struct for packaging together the functions for managing the daemon.
+/// Currently does not need any members, since most of the data is stored
+/// in files anyway.
+pub trait Daemon {
+    /// Daemon name, used for naming daemon files
+    const NAME: &'static str;
+
+    /// Helper function for getting the path to the daemon's working directory.
+    /// The directory is used for storing *.pid and *.socket files.
+    ///
+    /// # Returns
+    ///
+    /// Result containing the path of the daemons work directory.
+    fn get_wd_path() -> Result<PathBuf>;
```

Comment:
nitpick: I guess this function name is a bit confusing now, after refactoring. As it is just returning `/run/` or `/var/run`, and does not really have anything to od with working directory ;). Maybe lets rename it, for example: `get_runtime_data_directory()` or similar.